# 1. Data Collection

We are using 3 datasets in the training process:
* CelebA dataset
* Women faces with hijab
* The dataset we have collected from the university students

In [5]:
import sys
sys.path.append("path/to/UTILITY SCRIPTS/utility")

import utility  # Now you can use utility.py functions

In [6]:
from utility.utility import datacleaner

## 1.1 Collecting Student faces
Here we are collect the students data and organize it in a directory

## 1.2 Data Cleaning

### CelebA dataset

In [7]:
CELEBA_PATH = "/kaggle/input/face-vae/img_align_celeba/img_align_celeba"

In [8]:
# Checking the extentions in the dataset
non_jpg = datacleaner.extention_checker(CELEBA_PATH)

100%|██████████| 202599/202599 [14:11<00:00, 237.98it/s]

Number of JPG images found: 202599
Number of non-JPG images found: 0


[CelebA](https://mmlab.ie.cuhk.edu.hk/projects/CelebA.html) dataset has 202599 JPG images and 0 non-JPG.
I trust this dataset it is well known good dataset and after some research about the dataset on the internet I concluded that they are all high quality images all with the same size (178, 218).

### Hijab dataset
We went to look for hijab dataset because the first dataset has few hijab images or not at all.
So we need to cure the imbalance and this biased model. We found small dataset but we need to check it first.

In [12]:
HIJAB_PATH = "/kaggle/input/women-faces-with-hijabscientific-use-only"

In [13]:
# Checking the extentions in the dataset
non_jpg = datacleaner.extention_checker(HIJAB_PATH)

100%|██████████| 112/112 [00:00<00:00, 403.81it/s]

Number of JPG images found: 112
Number of non-JPG images found: 0


We check for the extentions and now we need to check for the quality of the dataset. We can plot some random sample of 10 images.

## 1.3 Face detection and cropping

## 1.4 Data preprocessing